docs: updated installation guides

pd93 opened the pull request:
Standardises the layout for all installation methods, adds labels and links to sources where possible. Also reorders the list with the official methods at the top and community owned ones at the bottom.

Also closes the longstanding #935.

andreynering reviewed (approved):
Great, this page needed an overhaul 👏 

The only small thing I noticed that would be nice to fix if easy, is that link anchors for the titles that have badges now have extra dashes:

![image](https://github.com/user-attachments/assets/c5c3bd9d-495d-4b9d-847f-9d70749607f1)

This means that existing links for these anchors will be broken, and new links will look slightly uglier.

pd93 commented:
> link anchors for the titles that have badges now have extra dashes:

Yeah, I noticed this just as you commented! Thankfully Docusaurus allows you to [manually set the anchor ids](https://docusaurus.io/docs/next/markdown-features/toc#heading-ids), so I have updated them all to match the previous values now.

I have also added the `pypi` package as discussed in #935.
